src/examples/librpmem/fibonacci/fibonacci.c

```diff
@@ -0,0 +1,403 @@
+ *       notice, this list of conditions and the following disclaimer in
+ *       the documentation and/or other materials provided with the
+ *       distribution.
+ *
+ *     * Neither the name of the copyright holder nor the names of its
+ *       contributors may be used to endorse or promote products derived
+ *       from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/*
+ * fibonacci.c -- fibonacci sequence generator for librpmem
+ */
+#include <assert.h>
+#include <errno.h>
+#include <time.h>
+#include <unistd.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <libpmem.h>
+#include <librpmem.h>
+
+#define POOL_SIGNATURE	"FIBO"
+#define FIBO_SIG_LEN RPMEM_POOL_HDR_SIG_LEN
+#define SIZE_1MB	(1 * 1024 * 1024)
+
+struct fibo_t {
+	char signature[FIBO_SIG_LEN];
+	unsigned n;	/* index */
+	uint64_t fn;	/* F_{n} */
+	uint64_t fn1;	/* F_{n + 1} */
+	int checksum;
+};
+
+#define POOL_SIZE	(size_t)(32 * 1024 * 1024)
+#define DATA_OFF	4096 /* rpmem header size */
+#define NLANES		4
+#define DATA_SIZE	(sizeof(struct fibo_t))
+
+#define GET_FIBO(pool)	((struct fibo_t *)(pool + DATA_OFF))
+
+#define BROKEN_LOCAL	(1 << 0)
+#define BROKEN_REMOTE	(1 << 0)
+
+static void parse_args(int argc, char *argv[], const char **target,
+		const char **poolset, const char **path);
+static void *map_pmem(const char *path, size_t *mapped_len);
+
+static RPMEMpool *remote_create_or_open(const char *target, const char *poolset,
+		void *pool, int *created);
+static int remote_write(RPMEMpool *rpp);
+static int remote_read(RPMEMpool *rpp, void *buff);
+
+static void fibo_cleanup(RPMEMpool *rpp, void *pool, struct fibo_t *fibo_r,
+		int *initialized);
+static void fibo_generate(void *pool);
+static void fibo_print(void *pool);
+
+int
+main(int argc, char *argv[])
+{
+	const char *target, *poolset, *path;
+	parse_args(argc, argv, &target, &poolset, &path);
+
+	struct fibo_t fibo_r; /* copy from remote */
+	int created; /* remote: 1 == created, 0 == opened */
+	int initialized; /* sequence: 1 == initialized, 0 = con be continued */
+	int ret;
+
+	/* map local pool */
+	size_t mapped_len;
+	void *pool = map_pmem(path, &mapped_len);
```

Comment:
I think we should use a typed variable here and modify all functions to accept the typed pool object.